### DISCUSSION TITLE
[Bug] Items in the cart do not seem to reflect whats in the checkout?

### DISCUSSION THREAD

**Discussion by maddalax** [Shopify/storefront-api-feedback#16] [Help] [answered]
I have 3 items in my cart, but when I go to the checkout URL, it only shows 2 items. I think it is showing items from an earlier checkout that I was testing. 

I attached a video of what is happening. I console.log my cart and showed the object has 3 line items, but once I go to the checkoutURL there's only 2.

https://user-images.githubusercontent.com/11528924/125168889-51809c00-e16d-11eb-8093-1d65799967b7.mov

**Comment by maddalax** [comment, marked as answer]
bump

**Reply by morganmccunn** [reply]
Sorry for the delayed reply here @maddalax - We'll look into this. Thank you for sharing this find!

**Reply by morganmccunn** [reply]
One thing to double check - are all the products you're testing with currently published to the Online Store as well? It's currently a limitation that the products need to be published to the online store as well as your headless app. Please confirm!

**Reply by maddalax** [reply]
@morganmccunn Yes they are published. If it helps, I've been able to work around this by making a brand new cart each time I add/remove an item, (instead of adding/removing items on the current cart). 

Making a brand new cart using 'cartCreate' mutation and passing all the items seems to work well.

**Reply by morganmccunn** [reply]
Okay thanks for confirming! Sounds like a terrible (but creative!) workaround though. We'll need to try to replicate and address this.

**Reply by cocoahero** [reply]
> I have 3 items in my cart, but when I go to the checkout URL, it only shows 2 items. I think it is showing items from an earlier checkout that I was testing.

@maddalax do you mean that you visited the cart's `checkoutUrl` when the cart only had 2 items, and then added the third?

**Reply by maddalax** [reply]
> > I have 3 items in my cart, but when I go to the checkout URL, it only shows 2 items. I think it is showing items from an earlier checkout that I was testing.
> 
> @maddalax do you mean that you visited the cart's `checkoutUrl` when the cart only had 2 items, and then added the third?

Yeah I added 2 items, visited the checkoutUrl, it looked good. 
I then went back, added a third item, visited checkoutUrl, and it only had the original 2 items, but the JSON data of the cart showed it has 3 items.

**Reply by cocoahero** [reply]
👍🏼  thanks for that confirmation - I have successfully reproduced the bug and we will try and get a fix out as soon as possible. Thanks again for the flag!

**Reply by maddalax** [reply]
Awesome!!

**Reply by cocoahero** [reply]
Hey @maddalax - the fix for this has been deployed! Please let us know if you're still seeing issues with it.
